bpf/mb_connect.c

```diff
@@ -38,8 +37,10 @@ static inline int udp4_connect(struct bpf_sock_addr *ctx)
     __u64 uid = bpf_get_current_uid_gid() & 0xffffffff;
     if (bpf_htons(ctx->user_port) == 53 && uid != SIDECAR_USER_ID) {
         // needs rewrite
-        struct origin_info origin = {.ip = ctx->user_ip4,
-                                     .port = ctx->user_port};
+        struct origin_info origin;
+        __builtin_memset(&origin, 0, sizeof(origin));
```

Comment:
This pattern avoids padding issues, otherwise there would be a verifier error
See https://github.com/iovisor/bcc/issues/2623#issuecomment-560214481